packages/astro/src/@types/astro.ts

```diff
@@ -1617,6 +1617,7 @@ export interface SSRMetadata {
 	renderers: SSRLoadedRenderer[];
 	pathname: string;
 	hasHydrationScript: boolean;
+	hasSolidHydrationScript: boolean;
```

Comment:
Rather than have framework-specific properties in this internal Result object, this should happen within the renderer itself. You can do this by using the Result object as a weakmap key. We do this in some of the integrations already. See for example in the Preact integration: https://github.com/withastro/astro/blob/main/packages/integrations/preact/src/server.ts#L47

Reply:
Thank you. I found the equivalent `getContext` in the Solid integration and changed it to use that. In the process I discovered another issue with the PR so I have temporarily changed it to "draft" status until I solve that issue.